### PR TITLE
BLUE-2003 extend orbit with gps configuration

### PR DESCRIPTION
**Pull request by coutinol**
need https://github.com/leanspace/orbits-repository/pull/146 to be merged first

### CODE REVIEW

File: services/orbits/orbits/parsers.go [hunk 1]
```diff
@@ -73,3 +113,39 @@ func (idealOrbit *IdealOrbit) FromMap(idealOrbitMap map[string]any) error {
...
+	return nil
+}
+
+func (gpsMetrics *GpsMetrics) FromMap(gpsMetricsMap map[string]any) error {
+	gpsMetrics.MetricIdForPositionX = gpsMetricsMap["metric_id_for_position_x"].(string)
+	gpsMetrics.MetricIdForPositionY = gpsMetricsMap["metric_id_for_position_y"].(string)
+	gpsMetrics.MetricIdForPositionZ = gpsMetricsMap["metric_id_for_position_z"].(string)
+	gpsMetrics.MetricIdForVelocityX = gpsMetricsMap["metric_id_for_position_x"].(string)
+	gpsMetrics.MetricIdForVelocityY = gpsMetricsMap["metric_id_for_position_y"].(string)
+	gpsMetrics.MetricIdForVelocityZ = gpsMetricsMap["metric_id_for_position_z"].(string)
```
Comment (jerome-leanspace) [inline services/orbits/orbits/parsers.go:141]: velocity not position

File: services/orbits/orbits/parsers.go [hunk 2]
```diff
@@ -38,6 +41,37 @@ func (idealOrbit *IdealOrbit) ToMap() map[string]any {
...
+}
+
+func (gpsMetrics *GpsMetrics) ToMap() map[string]any {
+	gpsMetricsMap := make(map[string]any)
+	gpsMetricsMap["metric_id_for_position_x"] = gpsMetrics.MetricIdForPositionX
+	gpsMetricsMap["metric_id_for_position_y"] = gpsMetrics.MetricIdForPositionY
+	gpsMetricsMap["metric_id_for_position_z"] = gpsMetrics.MetricIdForPositionZ
+	gpsMetricsMap["metric_id_for_position_x"] = gpsMetrics.MetricIdForVelocityX
+	gpsMetricsMap["metric_id_for_position_y"] = gpsMetrics.MetricIdForVelocityY
+	gpsMetricsMap["metric_id_for_position_z"] = gpsMetrics.MetricIdForVelocityZ
```
Comment (jerome-leanspace) [inline services/orbits/orbits/parsers.go:62]: velocity not position

File: testing/orbits/orbits/main.tf [hunk 3]
```diff
@@ -11,6 +11,36 @@ variable "satellite_id" {
...
+variable "metric_id_for_velocity_x" {
+  type        = string
+  description = "The ID of the metric which hold gps velocity X data."
+}
+
+variable "metric_id_for_velocity_y" {
+  type        = string
+  description = "The ID of the metric which hold gps velocity Y data."
+}
+
```
Comment (jerome-leanspace) [inline testing/orbits/orbits/main.tf:38]: Please add them in the testing/main.tf as a parameter otherwise it won't work

**Review by jerome-leanspace — COMMENTED** [review]
You've added this in testing but not in example

**Comment by coutinol** [comment]
finding & fixing some other issues

File: services/orbits/orbits/schemas.go [hunk 4]
```diff
@@ -34,6 +34,14 @@ var orbitSchema = map[string]*schema.Schema{
...
 		},
 	},
+    "gps_configuration": {
+        Type:     schema.TypeList,
+        Optional: true,
+        MinItems: 1,
+        MaxItems: 1,
+        Elem: &schema.Resource{
+            Schema: gpsConfigurationSchema,
+        },
```
Comment (elodietinland) [inline services/orbits/orbits/schemas.go:44]: Missing } here to close the gps_configuration list

**Comment by coutinol** [comment]
fixed & tested on develop/yuri -> see output 652d86f3-7022-4e64-b379-c99a4fc5d136

**Review by elodietinland — APPROVED** [review]

**Comment by jerome-leanspace** [comment]
FYI It's still missing the new field in the examples folder

**Comment by elodietinland** [comment]
> FYI It's still missing the new field in the examples folder

Done in https://github.com/leanspace/terraform-provider-leanspace/pull/184

**Pull request merged**